### DISCUSSION TITLE
Error calling minio API: The specified bucket does not exist

### DISCUSSION THREAD

**Discussion by leipanhz** [minio/minio#20596] [Q&A]
I set up minio on Mac using homebrew.  I can access UI and create bucket there. However, when I tried with adding/listing buckets in go, as instructed in: https://github.com/minio/minio-go?tab=readme-ov-file, there is following error:
 The specified bucket does not exist.

Changed bucket visibility from private to public, that didn't help. I have able to list and create bucket from the sample minio: "play.min.io".  

$ minio --version
minio version RELEASE.2024-10-13T13-34-11Z (commit-id=d10bb7e1b667c2df72c394ef1fa52ab4a6802d0f)
Runtime: go1.22.8 darwin/arm64
License: GNU AGPLv3 - https://www.gnu.org/licenses/agpl-3.0.html
Copyright: 2015-2024 MinIO, Inc.

**Comment by marktheunissen** [comment]
Can you run this sample code from here: https://github.com/minio/minio-go/blob/master/examples/s3/getobject.go ?

If it fails, please paste all the environment variables and the output.

If I was to take a wild guess it's probably something to do with host-style vs. path-style bucket referencing, have you changed that?

**Comment by leipanhz** [comment]
@marktheunissen Thank you for looking into this. The following is the output (I added some logs before and after calling GetObject:

Calling GetObject. Monday, 25-Nov-24 11:13:32 PST
After GetObject with no error. Monday, 25-Nov-24 11:13:32 PST
2024/11/25 11:13:32 Cannot read Stat The specified bucket does not exist.

I have verified that I can access the bucket via the minio GUI with default password (secure set to false). From the code GetObject has no error, I can read Stat or fetch bucket to the localfile. 

What environment variables do you want to check? I don't have any minio related env being Set. The experiment is set up on Mac.

**Comment by marktheunissen** [comment]
`The specified bucket does not exist.` ... does the bucket exist?

**Reply by leipanhz** [reply]
Yes it does exist. I can view data in the bucket via Minio UI and also GetObject in the test code you provided was successful. Error is from here:

stat, err := reader.Stat()
	if err != nil {
		log.Fatalln("Cannot read Stat", err)
	}

Data can't be retrieved to local file

**Reply by leipanhz** [reply]
Code is in the following:


reader, err := s3Client.GetObject(context.Background(), "data", "data-arrow-1", minio.GetObjectOptions{})
	if err != nil {
		log.Fatalln("Cannot get bucket data", err)
	} else {
		print("After GetObject with no error")
		fmt.Println(time.Now().Format(time.RFC850))
	}
	defer reader.Close()
	localFile, err := os.Create("my-testfile")
	if err != nil {
		log.Fatalln("Cannot create my-testfile", err)
	}
	defer localFile.Close()
	stat, err := reader.Stat()
	if err != nil {
		log.Fatalln("Cannot read Stat", err)
	}

**Reply by marktheunissen** [reply]
Works fine for me... 

```
❯ mc ls local/helloworld/video.mp4
[2024-11-15 16:19:55 AEDT] 149MiB STANDARD video.mp4
```

main.go

```
package main

import (
	"context"
	"fmt"
	"log"
	"os"
	"time"

	"github.com/minio/minio-go/v7"
	"github.com/minio/minio-go/v7/pkg/credentials"
)

func main() {
	s3Client, err := minio.New("127.0.0.1:9000", &minio.Options{
		Creds: credentials.NewStaticV4("minioadmin", "minioadmin", ""),
	})
	if err != nil {
		log.Fatalln("minio.New", err)
	}

	reader, err := s3Client.GetObject(context.Background(), "helloworld", "video.mp4", minio.GetObjectOptions{})
	if err != nil {
		log.Fatalln("Cannot get bucket data", err)
	} else {
		print("After GetObject with no error")
		fmt.Println(time.Now().Format(time.RFC850))
	}
	defer reader.Close()
	localFile, err := os.Create("my-testfile")
	if err != nil {
		log.Fatalln("Cannot create my-testfile", err)
	}
	defer localFile.Close()
	stat, err := reader.Stat()
	if err != nil {
		log.Fatalln("Cannot read Stat", err)
	}
	fmt.Println(stat.Size)
}
```

Prints:

```
After GetObject with no errorTuesday, 26-Nov-24 10:45:19 AEDT
156638187
```

where 156638187 is the correct size of the object `video.mp4`.

Is it something to do with the homebrew configuration? Paste output of `mc admin info ALIAS --json`

**Reply by leipanhz** [reply]
I can put and view the bucket via mc:
`$ mc ls myminio/prefetchdata/controller.log
[2024-11-26 10:29:09 PST] 3.9KiB STANDARD controller.log`

I don't have mc admin set up. Is it needed? Earlier I uploaded bucket via minio UI
$ mc admin info ALIAS --json
{
 "status": "error",
 "error": {
  "message": "Unable to initialize admin connection.",
  "cause": {
   "message": "No valid configuration found for 'ALIAS' host alias",
   "error": {}
  },
  "type": "fatal"
 }
}
